extensions/mailer/deployment/src/main/java/io/quarkus/mailer/deployment/MailerProcessor.java

```diff
@@ -75,18 +74,17 @@ void registerAuthClass(BuildProducer<ReflectiveClassBuildItem> reflectiveClass)
 
     @BuildStep
     @Record(ExecutionTime.RUNTIME_INIT)
-    MailerBuildItem build(BuildProducer<FeatureBuildItem> feature, MailConfigRecorder recorder, VertxBuildItem vertx,
-            BeanContainerBuildItem beanContainer, LaunchModeBuildItem launchMode, ShutdownContextBuildItem shutdown,
-            MailConfig config) {
+    void build(BuildProducer<FeatureBuildItem> feature, MailConfigRecorder recorder,
+            MailConfig config, LaunchModeBuildItem launchMode,
+            BuildProducer<SyntheticBeanBuildItem> syntheticBeanBuildItemBuildProducer) {
 
         feature.produce(new FeatureBuildItem(FeatureBuildItem.MAILER));
 
-        RuntimeValue<MailClient> client = recorder.configureTheClient(vertx.getVertx(), beanContainer.getValue(), config,
-                shutdown);
-
-        recorder.configureTheMailer(beanContainer.getValue(), config, launchMode.getLaunchMode());
-
-        return new MailerBuildItem(client);
+        syntheticBeanBuildItemBuildProducer.produce(SyntheticBeanBuildItem.configure(MailerSupport.class)
+                .scope(Singleton.class)
+                .supplier(recorder.mailerSupportSupplier(config, launchMode.getLaunchMode()))
```

Comment:
AFAIK it's possible to inject both the config object and the `LaunchMode` so there is no need to register a synthetic bean, or?

Reply:
Ah yes, I forgot that `LaunchMode` is also a bean, excellent!